pennylane/templates/subroutines/hilbert_schmidt.py

```diff
@@ -268,9 +268,10 @@ def compute_decomposition(
         )
 
         # Unitary U
-        for op_u in u_tape.operations:
-            qml.apply(op_u)
-            decomp_ops.append(op_u)
+        if qml.QueuingManager.recording():
+            decomp_ops.extend(qml.apply(op_u) for op_u in u_tape.operations)
+        else:
+            decomp_ops.extend(u_tape.operations)
```

Comment:
real change alert! I can undo this here if needed, but the docstring failed because we tried to apply outside of a context

Reply:
I'm breaking this change into a separate PR so we can link it as evidence in the technical roadmap.

Reply:
#4719 merged! file is now untouched in this PR